File: frontend/src/services/expenseService.ts
import api from './api';

export const expenseService = {
  getAll: (params?: Record<string, string>) => api.get('/expenses', { params }),
  getById: (id: string) => api.get(`/expenses/${id}`),
  create: (data: Record<string, unknown>) => api.post('/expenses', data),
  update: (id: string, data: Record<string, unknown>) => api.put(`/expenses/${id}`, data),
  delete: (id: string) => api.delete(`/expenses/${id}`),
  getByCategory: (category: string) => api.get(`/expenses/category/${category}`),
  getMonthly: (year: number, month: number) => api.get(`/expenses/monthly/${year}/${month}`),
};

File: backend/src/main/java/com/finwise/service/ExpenseService.java
package com.finwise.service;

import com.finwise.dto.expense.ExpenseRequest;
import com.finwise.dto.expense.ExpenseResponse;
import java.util.List;

public interface ExpenseService {
    List<ExpenseResponse> getAllByUser(String userId);
    ExpenseResponse getById(String id);
    ExpenseResponse create(String userId, ExpenseRequest request);
    ExpenseResponse update(String id, ExpenseRequest request);
    void delete(String id);
    List<ExpenseResponse> getByCategory(String userId, String category);
    List<ExpenseResponse> getByMonth(String userId, int year, int month);
}
